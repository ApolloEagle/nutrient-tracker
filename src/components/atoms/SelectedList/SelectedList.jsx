import {
  List,
  ListItem,
  ListItemText,
  Collapse,
  IconButton,
} from "@mui/material";
import { TransitionGroup } from "react-transition-group";

function SelectedList({ selectedFoods, handleDelete, deleteIcon }) {
  return (
    <List>
      <TransitionGroup>
        {!!selectedFoods &&
          selectedFoods.map((selectedFood, index) => {
            return (
              <Collapse key={index}>
                <ListItem
                  sx={{
                    bgcolor: "primary.light",
                    marginBottom: 1,
                    borderRadius: 2,
                    color: "white",
                  }}
                  secondaryAction={
                    <IconButton
                      edge="end"
                      onClick={() => handleDelete(selectedFood)}
                    >
                      {deleteIcon}
                    </IconButton>
                  }
                >
                  <ListItemText>{selectedFood.description}</ListItemText>
                </ListItem>
              </Collapse>
            );
          })}
      </TransitionGroup>
    </List>
  );
}

export default SelectedList;
