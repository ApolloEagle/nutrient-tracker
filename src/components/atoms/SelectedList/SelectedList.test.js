import React from 'react';
import { shallow } from 'enzyme';
import SelectedList from './SelectedList';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const wrapper = shallow(<SelectedList />);

describe('Selected List', () => {
  it('Renders selected list', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
