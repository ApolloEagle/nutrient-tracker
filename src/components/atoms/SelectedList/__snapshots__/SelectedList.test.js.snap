// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selected List Renders selected list 1`] = `
<ForwardRef(List)>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  />
</ForwardRef(List)>
`;
